packages/beacon-node/src/network/processor/gossipHandlers.ts

```diff
@@ -312,7 +321,13 @@ function getDefaultHandlers(modules: ValidatorFnsModules, options: GossipHandler
       }
       const blockInput = await validateBeaconBlob(signedBlob, serializedData, topic.index, peerIdStr, seenTimestampSec);
       if (blockInput !== null) {
-        handleValidBeaconBlock(blockInput, peerIdStr, seenTimestampSec);
+        // TODO DENEB:
+        //
+        // With blobsPromise the block import would have been attempted with the receipt of the block gossip
+        // and should have resolved the availability promise, however we could track if the block processing
+        // was halted and requeue it
+        //
+        // handleValidBeaconBlock(blockInput, peerIdStr, seenTimestampSec);
```

Comment:
So the data received through the blob_sidecar topic is discarded? Both if branches do nothing

Reply:
Looking into the blob validation function noticed that there's an unnecessary call to regen here

https://github.com/ChainSafe/lodestar/blob/fa30bcf10086292883ef086202474ac4df67ffe1/packages/beacon-node/src/chain/validation/blobSidecar.ts#L103-L107

you only need the global pubkey cache and a config, no need to calculate a specific state.

Reply:
> So the data received through the blob_sidecar topic is discarded? Both if branches do nothing

yes currently only seeing block only we trigger processing doesn't matter if all blobs are available or not (we concurrently wait for availability while other aspects of block are getting verified) and if the data doesn't become available in a threshold time we throw error which triggers pulling the block/blobs by root.

We can also threshold here the delay from the starting of the slot to throw the unknown block event. if the block was there (then we get non null blockinput) it must already be in the block processing queue waiting for blobs to become available which this blob would have resolved the promise if this was the last blob needed.

But if that was not the case then we can throw unknown block to pull by root.

Reply:
> you only need the global pubkey cache and a config, no need to calculate a specific state.

ahh right, had this copied over from block, will see how to optimize.

Reply:
updated in the tracker (https://github.com/ChainSafe/lodestar/issues/5279) to tackle on top of blob inclusion proof PR